source/common/router/router.cc

```diff
@@ -1113,6 +1126,14 @@ void Filter::UpstreamRequest::encodeData(Buffer::Instance& data, bool end_stream
 
     buffered_request_body_->move(data);
   } else {
+    // Encodes metadata if exists.
```

Comment:
When does this happen here and below? In the case where we don't have an encoder, won't we deal with any pending metadata when the pool is ready? Can we instead just assert that we don't have any pending metadata here and below?

Reply:
Good catch! Removed and added asserts.